### ISSUE TITLE
KeySet class in python is missing methods

### ISSUE THREAD

**Issue by dwiel** [borglab/gtsam#1606]
## Feature

The `KeySet` class in python should have more methods available like `__iter__`, `__contains__`, `remove`, etc.

## Motivation

I'd like to be able to do things like:

```
for key in graph.keys():
    if key not in initial_estimate.keys():
        print(f"WARNING: missing key {key} in initial_estimate")
```

**Comment by ProfFan** [comment]
Yeah this would be a great extension. We may need some special code to do this though, as `KeySet` is a bit more than a standard `std::set`. CC @varunagrawal

**Comment by varunagrawal** [comment]
I think the easiest way to achieve this is to replace `gtsam::KeySet` with `std::set<gtsam::Key>` (aka the superclass) in all the .i files?

**Comment by varunagrawal** [comment]
A better way to do this is to integrate dunder methods (e.g. `__iter__`) into our wrapper library `gtwrap`. @dwiel can you compile a list of all the special methods needed?

**Comment by varunagrawal** [comment]
Accidentally closed this. This issue will be resolved when we merge #1712.
